### DISCUSSION TITLE
Live Update: Implementing Semver Validation for Bundle Updates

### DISCUSSION THREAD

**Discussion by luisbytes** [capawesome-team/capacitor-plugins#255] [Ideas]
Currently, no `versionName` is assigned to bundles, which creates an issue, as it’s not possible to easily identify bundles uploaded to the Cloud without an assigned version.

Additionally, there's a problematic scenario due to the lack of `versionName` validation. For instance, version 2.3.0 might be available both in the store and as a live update. This could lead to a user installing version 2.3.0 from the store and then unnecessarily downloading the same version from the live update.

The ideal solution would be to enable the plugin to accept semver and implement the necessary validations for downloading updates.

For example, a new live update should only be downloaded and activated via the `sync` method or manually through the `downloadBundle` method if its version is greater than the current binary version.

**Comment by robingenz** [comment]
Please have a look at [this comment](https://github.com/capawesome-team/capacitor-plugins/discussions/254#discussioncomment-10300907) regarding semantic versioning in the Live Update plugin.

> This could lead to a user installing version 2.3.0 from the store and then unnecessarily downloading the same version from the live update.

If you use [Versioned Channels](https://capawesome.io/blog/how-to-restrict-capacitor-live-updates-to-native-versions/#versioned-channels), then there is no reason to release a bundle as a live update that is already included in the native version.
